Added a new field for world countries dataset with the ISO numeric code

jsanz opened the pull request:
Fixes #211 

This PR adds a new table in the `sources/world` folder with the list of iso 2, 3, and numeric country codes, adding the numeric code to the world countries GeoJSON and TopoJSON targets in the Makefile.

CSV is taken from <https://www.iso.org/obp> website, more details in ISO [country codes landing page](https://www.iso.org/iso-3166-country-codes.html).

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+ems-file-service+pull-request/124/)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+ems-file-service+pull-request/125/)

nickpeihl commented on sources/world/Makefile:
```diff
@@ -123,24 +123,25 @@ clean: ## Remove output files
   -filter "[$${codesA},$${codesB}].indexOf(region_iso_code) == -1" \
   -dissolve 'country_iso2_code' 'copy-fields=country_iso3_code,country_name' \
   -rename-fields "iso2=country_iso2_code,iso3=country_iso3_code,name=country_name" \
+  -join ./country-codes.csv "keys=iso2,iso2" "fields=iso_numeric" \
```
I think we need to add the `iso_numeric` codes as strings rather than integers. Maybe using the `string-fields` option in the [Mapshaper `-join`](https://github.com/mbloch/mapshaper/wiki/Command-Reference#-join)?

nickpeihl commented on sources/world/country-codes.csv:
```diff
@@ -0,0 +1,250 @@
+iso2,iso3,iso_numeric
+AF,AFG,4
```
The [ISO numeric codes are three digits](https://en.wikipedia.org/wiki/ISO_3166-1_numeric) so some countries will need leading zeros. 

For example: 
```suggestion
AF,AFG,004
```

jsanz replied:
Yep, sorry about that, thanks for the pointer!

nickpeihl commented on data/world_countries_v7.topo.json:
```diff
@@ -15,6 +15,7 @@
           "properties": {
             "iso2": "AD",
             "iso3": "AND",
+            "iso_numeric": 20,
```
I think these should be strings.

For example
```suggestion
            "iso_numeric": "020",
```

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+ems-file-service+pull-request/126/)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+ems-file-service+pull-request/127/)

jsanz commented:
@nickpeihl I've addressed the leading zeros issue.

I also removed unnecessary attribution to World Bank and Wikidata that was introduced when metrics were added (I guess?) and added attribution to OSM as we have for the Admin Regions dataset since world borders are derived from Admin Regions.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+ems-file-service+pull-request/128/)

nickpeihl reviewed (approved):
lgtm!
